datafusion/physical-plan/src/test/exec.rs

```diff
@@ -614,6 +614,103 @@ impl ExecutionPlan for StatisticsExec {
     }
 }
 
+/// A mock execution plan that returns per-partition statistics
```

Comment:
Maybe you can just create `DataSourceExec` with statistics to do tests. https://github.com/apache/datafusion/blob/main/datafusion/core/tests/physical_optimizer/partition_statistics.rs#L189